lib/rpc/rpc_mainchain.dart

```diff
@@ -1,30 +1,80 @@
+import 'dart:async';
+
 import 'package:dart_coin_rpc/dart_coin_rpc.dart';
 import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:sidesail/pages/tabs/settings_tab.dart';
+import 'package:sidesail/rpc/rpc.dart';
 import 'package:sidesail/rpc/rpc_config.dart';
 
 /// RPC connection to the mainchain node.
-abstract class MainchainRPC {
+abstract class MainchainRPC extends ChangeNotifier implements RPCConnection {
```

Comment:
Getting heavy Java-vibes from certain parts of Dart! I haven't seen this much class inheritance since my Scala days in 2019